---
title: "Untitled"
author: "Gabe Runte"
date: "5/26/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(phyloseq)
library(vegan)
library(here)
library(janitor)
library(gridExtra)
library(dada2)
```


```{r}
sos_consensus = read_csv(here("data", "sos_consensus1.csv")) %>% 
  clean_names() %>% 
  dplyr::select(name, sequence) %>% 
  rename(sequence_list_name = name)

sos_tips = read_csv(here("data", "sos_contigs1.csv")) %>% 
  clean_names() %>% 
  filter(!is.na(percent_gc)) %>% 
  separate(name, c("sos", "core_id", "plate_number", "column", "row")) %>% 
  mutate(plate_number = as.numeric(plate_number))%>% 
  mutate(column = as.numeric(column)) %>% 
  mutate(sos_core_id = paste(sos, core_id, sep = "-")) %>% 
  relocate(sos_core_id) %>% 
  dplyr::select(1,3:6, sequence_list_name) %>% 
  left_join(sos_consensus)
  

  
master = read.csv(here("data", "Datasheets_and_Templates - SoilCoreMetadata.csv")) %>% 
  clean_names()

sample_locations = read_csv(here("data", "sos_tippulling_sequencing - sample_locations.csv")) %>% 
  rename(sos_core_id = sample) %>% 
  mutate(plate_number = as.numeric(plate_number))%>% 
  mutate(column = as.numeric(column)) %>% 
  dplyr::select(!notes)


sanger_sheet = sos_tips %>% 
  left_join(sample_locations)
```


```{r, eval = FALSE}
unite.ref <- "/Users/Gabe/Desktop/phd/ghecto/unite_db/sh_general_release_s_10.05.2021/sh_general_release_dynamic_s_10.05.2021.fasta"



tiptaxa = assignTaxonomy(sanger_sheet$sequence, unite.ref, multithread = TRUE, tryRC = TRUE)

sos_tiptaxa = tibble(as.data.frame(tiptaxa)) %>% 
  mutate(sos_core_id = sanger_sheet$sos_core_id) %>% 
  mutate(plate_number = sanger_sheet$plate_number) %>% 
  mutate(column = sanger_sheet$column) %>% 
  mutate(row = sanger_sheet$row)

write_csv(sos_tiptaxa, here("data", "sos_tiptaxa_sanger_contigs1.csv"))
```

```{r}
 sos_tiptaxa = read_csv(here("data", "sos_tiptaxa_sanger_contigs1.csv"))

morph = sanger_sheet %>% 
               dplyr::select(sequence, sos_core_id, plate_number, column, row, morphotype) %>% 
  left_join(sos_tiptaxa) %>% 
  left_join(sanger_sheet %>% 
              dplyr::select(!core_id)) 

#write_csv(morph, here("data", "sos_contig_morphotyping1.csv"))
```

